resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java

```diff
@@ -146,6 +147,10 @@ public class DnsNameResolverTest {
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(DnsNameResolver.class);
     private static final long DEFAULT_TEST_TIMEOUT_MS = 30000;
 
+    static {
+        System.setProperty("io.netty.resolver.dns.trycnameonaddresslookups", "true");
```

Comment:
is a new JVM spewed for every test? if not this may not have any impacts because `TRY_FINAL_CNAME_ON_ADDRESS_LOOKUPS` is cached. Consider adding a package private variable in `DnsNameResolver` to control the behavior for easier testing.